### ISSUE TITLE
versionCode not increased

### ISSUE THREAD

**Issue by IzzySoft** [DrMint/Litrato#4]
it looks like `versionCode` is never increased before releasing a new version. From the last release's [`build.gradle`](https://github.com/DrMint/Litrato/blob/1.11/app/build.gradle#L9):

```gradle
        versionCode 1
        versionName "1.1"
```

This is from tag `1.11` – the third release, so I'd expected

```gradle
        versionCode 3
        versionName "1.11"
```

instead. Unfortunately, that's not just a cosmetic thing: `versionCode` is what Android uses internally to tell versions apart. Which means, the newer versions won't be offered for update, as Android thinks they're already installed. And even if force-updated, it will make reporting of issues difficult: which "version 1.1" had the user installed – this one (1.11) or the previous one?

**TL;DR:** could you please increase `versionCode` with each release (for technical reasons) – and `versionName` (to not confuse users :speak_no_evil:)? Thanks!

**Comment by DrMint** [comment]
Hi IzzySoft, thank you again for commenting more than a year on this subject :)
I totally forgot about that. To correct this problem, do you think I should make new version with versionCode = versionName = GitHub release tag = 1.12?

**Comment by DrMint** [comment]
I just realized that versionCode should be an integer apparently. I'll do versionCode equals 2 and the rest equals 1.12

**Comment by IzzySoft** [comment]
Eh, so I remembered correctly I reported that already once… :see_no_evil: Yes, `versionCode` must be a positive integer. There are several patterns for this used in practice:

* simply use it as "counter" – increase by 1 each release, then look at it after 5 years and say: oh, we did 42 releases :grin:
* doing some maths. Using [semantic versioning](https://semver.org/) for `versionName`, then calculate `versionCode` as `10000xmajor + 100xminor + patch` – so e.g. v1.0.3 results in 10003
* lazy bob wants to be cool: use the time, pattern YYYYMMDD (err, limits you to 1 release per day – so don't make mistakes :rofl:)

Just remember: you can never ever decrease. So switching is possible from item 1 -> 3 but not the other way around.

> I'll do versionCode equals 2 and the rest equals 1.12

Works for starters, yes! Always look towards the future – as that's where the rest of your life will happen :smile:

PS: Should you like [a nice badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it), you're welcome [to link it](http://apt.qumran.org/fdroid/index/apk/com.example.retouchephoto) :smiley:

**Comment by IzzySoft** [comment]
Just noticed my "link it" was broken; fixed :wink:

**Comment by DrMint** [comment]
Oh, I never quite got what you meant with the badge thing. Yeah sure, I'll add it!

**Comment by IzzySoft** [comment]
> I never quite got what you meant with the badge thing.

Yeah, my bad, giving you the link to my test site which is not available from outside my test net :see_no_evil:

> Yeah sure, I'll add it!

Thanks! My repo seems to have a constantly growing number of users – 250k+ hits per day, 25k+ visitors per day downloading ~25G per day currently. For the close to 700 apps listed there it enables them to "set and forget": install the app and automatically get updates :smiley:
